botocore/tokens.py

```diff
@@ -324,3 +327,25 @@ def load_token(self):
         return DeferredRefreshableToken(
             self.METHOD, self._refresher, time_fetcher=self._now
         )
+
+    def _resolve_source_profile(self, profiles, initial_profile_config):
+        profile_name = initial_profile_config.get('source_profile', None)
+        source_profile = profiles.get(profile_name, {})
+        visited_profiles = {profile_name}
+        while source_profile.get('source_profile', None):
+            profile_name = source_profile.get('source_profile')
+            source_profile = profiles.get(profile_name, {})
+            if profile_name in visited_profiles:
+                valid = self._has_static_credentials(source_profile)
+                if valid:
+                    break
+                raise InfiniteLoopConfigError(
+                    source_profile=profile_name,
+                    visited_profiles=visited_profiles,
+                )
+            visited_profiles.add(profile_name)
+        return source_profile
+
+    def _has_static_credentials(self, profile_config):
```

Comment:
We should also be able to remove this logic if we do not raise errors for infinite loops in this auth token class.